### PR TITLE
go 1.17rc - testing only

### PR DESCRIPTION
**Pull request by andig**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

Testing PR to see how far we can get with go 1.17 as discussed on https://github.com/Homebrew/discussions/discussions/1889#discussioncomment-1050230

Created using:

    brew bump-formula-pr golang --version 1.17rc1 --force --no-audit

### CODE REVIEW

File: Formula/go.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Go < Formula
   desc "Open source programming language to build simple/reliable/efficient software"
   homepage "https://golang.org"
-  url "https://golang.org/dl/go1.16.6.src.tar.gz"
-  mirror "https://fossies.org/linux/misc/go1.16.6.src.tar.gz"
-  sha256 "a3a5d4bc401b51db065e4f93b523347a4d343ae0c0b08a65c3423b05a138037d"
+  url "https://golang.org/dl/go1.17rc1.src.tar.gz"
+  mirror "https://fossies.org/linux/misc/go1.17rc1.src.tar.gz"
```
Comment (carlocab) [inline Formula/go.rb:5]:
```suggestion
  mirror "https://fossies.org/linux/misc/go1.17rc1.src.tar.gz"
  version "1.17"
```

Comment (SMillerDev) [reply Formula/go.rb:5]: It also needs a whitelist

Comment (andig) [reply Formula/go.rb:5]: Still stuck at the same issue:

    Stable version URLs should not contain rc1

> It also needs a whitelist

Could you elaborate? Not sure what to do to advance CI beyond the audit check.

**Comment by andig** [comment]
Still no:

    Stable version URLs should not contain rc1

**Comment by andig** [comment]
Wow- looks as if CI is actually passing this time? In this case I'd suggest to keep the PR open and update on release or just close it and go through regular bump-formula on release.

Would replace https://github.com/Homebrew/homebrew-core/pull/81824, too.

**Comment by andig** [comment]
I have not looked through all of the errors, but there is a number of:

- Error: dep: no bottle available: not sure what to do here. Raise upstream issues for each occurrence or deprecate the formula? Imho- if repo has not upgraded to go modules yet it is a good indication that the project is inactive?
- Failed to install service files: what are these? Seem to be happening independentely from any compile errors?
- Some unexpected go version due to the rc, these should vanish with go final

**Comment by andig** [comment]
- [ ] mattermost: https://github.com/mattermost/mmctl/issues/363
- [ ] https://github.com/lucy/mpdviz: not updated in years -> deprecate?

**Comment by carlocab** [comment]
`Failed to install service files` should've been fixed in Homebrew/brew#11777. We could probably re-run CI to make sure.

**Comment by andig** [comment]
> We could probably re-run CI to make sure.

could we do that to get rid of the spurious errors? I can compile a full tasklist afterwards.

**Comment by carlocab** [comment]
CI isn't running for some reason. Let's try to get this going again...

**Comment by andig** [comment]
Quick analysis. There are test failures:

- [ ] akamai
- [ ] awsweeper
- [ ] bitrise
- [ ] darksky-weather
- [ ] func-e
- [ ] ooniprobe
- [ ] server-go
- [ ] storj-uplink
- [ ] tunnel
- [ ] xray

dep deprecations:

- [ ] docker-machine-driver-hyperkit
- [ ] ironcli
- [ ] vert

and finally SHA mismatches:

- [ ] baidupcs-go
- [ ] convox
- [ ] helmfile
- [ ] logcli
- [ ] promtail

**Comment by andig** [comment]
I will open deprecation PRs for anything using dep. These repos have not seen updates for >2 to 6 years.

**Review by BHPettersen — DISMISSED** [review]

**Comment by branchvincent** [comment]
`logcli` will be fixed by a new version #82894

**Comment by andig** [comment]
Can we get the dep deprecations merged?

File: Formula/go.rb [hunk 2]
```diff
@@ -60,6 +60,9 @@ def install
     # Remove useless files.
     # Breaks patchelf because folder contains weird debug/test files
     rm_rf Dir[libexec/"src/debug/elf/testdata"]
+
+    # Remove 32-bit test data
+    rm_rf Dir[libexec/"src/runtime/pprof/testdata/test32"]
```
Comment (carlocab) [inline Formula/go.rb:65]:
```suggestion
    (libexec/"src/runtime/pprof/testdata/test32").rmtree
```
From https://github.com/Homebrew/homebrew-core/pull/82830#discussion_r684678216

**Comment by andig** [comment]
We're 8 days away from the planned go 1.17 release. How shall we proceed with the SHA mismatches and test errors?

**Comment by andig** [comment]
Clsing in favor of 1.17 final

**Pull request closed without merging**